pennylane/workflow/qnode.py

```diff
@@ -1024,11 +1024,6 @@ def construct(self, args, kwargs):  # pylint: disable=too-many-branches
         params = self.tape.get_parameters(trainable_only=False)
         self.tape.trainable_params = qml.math.get_trainable_indices(params)
 
-        if any(isinstance(m, CountsMP) for m in self.tape.measurements) and any(

```

Comment:
Error in `jax_jit.py` is sufficient.  Comes slightly later, but it is the responsibility of the jax-jit interface, not the QNode.